lib/datasource/maven/index.js

```diff
@@ -11,8 +11,54 @@ module.exports = {
   getPkgReleases,
 };
 
+const cacheNamespace = 'datasource-maven';
+
+async function getPkgReleases({ lookupName, registryUrls, groupName, deps }) {
```

Comment:
Can you explain more about your design choice for groupName? I hadn't expected we need to pass groupName to the datasource. I would prefer if we could handle it using depName/lookupName and not add parameters like grouName/deps that are not shared with other datasources.

Reply:
I needed some key to handle grouping, as well as other dependencies in this group. Setting the `groupName` in extracting phase led to one PR per variable, instead of many PRs for each dependency. I found I can use this fields in datasource.

Actually I would like to implement the better design. At the extracting phase we can use `extractAllPackageFiles` and handle cross-file props. It would be better to handle common props in `updateDependency(content, upgrade)`, but I'm not sure I can use context there.